plugin/documents.py

```diff
@@ -152,7 +152,6 @@ def __del__(self) -> None:
         self._stored_region = sublime.Region(-1, -1)
         self._color_phantoms.update([])
         self.view.erase_status(AbstractViewListener.TOTAL_ERRORS_AND_WARNINGS_STATUS_KEY)
-        self._clear_highlight_regions()
```

Comment:
I removed this as I didn't see why there is a need to clear highlights in the `__del__` method. Should I put it back?

Reply:
It's so that when the plugin is disabled we clean up our trash (not everything gets cleaned up still, but we're getting there :) ) Please keep this line in the `__del__`.

Reply:
I added it back.
I wasn't aware that the `__del__` method was somehow related to the plugin_unload event. :)

Reply:
Session shutdown which can be trigger with "restart servers" also